### PR TITLE
Update the ES snapshot script to pass creds

### PR DESCRIPTION
**Pull request by rlvoyer**
This updates the ES snapshotting script to pass creds now that our Elasticsearch clusters require
authentication. Note that the ES snapshot image is used both by cetera and spandex, thus requiring
a generic parameter corresponding to credentials file downloaded via clortho.

These changes were tested locally against the Spandex cluster in staging.

~~- [ ] styleCheck passes~~
~~- [ ] unit tests pass~~
~~- [ ] integration tests pass~~
- [x] manual tests pass
~~- [ ] API docs updated~~

The corresponding tasks-chronos PR: https://github.com/socrata/tasks-chronos/pull/143

### CODE REVIEW

File: elasticsearch-snapshotter/docker/ship.d/run [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
 
-exec su socrata -c "exec  /etc/ship.d/snapshot.sh"
+set -e
+
+if [ -z "$SNAPSHOT_ID" ]; then
+    SNAPSHOT_ID=$(date +%Y%m%d%H%M%S)
+fi
+
+curl -s --fail -H 'Content-Type: application/json' -u $ES_USERNAME:$ES_PASSWORD -XPUT http://$ES_HOST:$ES_PORT/_snapshot/$SNAPSHOT_NAME \
+     -d "{ \"type\": \"s3\", \"settings\": { \"bucket\": \"$SNAPSHOT_NAME\", \"server_side_encryption\": true } }"
```
Comment (aynleslie) [inline elasticsearch-snapshotter/docker/ship.d/run]: do we no longer need to specify the `AWS_REGION`?

File: elasticsearch-snapshotter/docker/ship.d/run [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+echo
+
+curl -s --fail -H 'Content-Type: application/json' -u $ES_USERNAME:$ES_PASSWORD -XPUT http://$ES_HOST:$ES_PORT/_snapshot/$SNAPSHOT_NAME/$SNAPSHOT_ID \
+     -d "{ \"indices\": \"$INDICES\", \"ignore_unavailable\": $IGNORE_UNAVAILABLE, \"include_global_state\": $INCLUDE_GLOBAL_STATE }"
+echo
+
+response=$(curl -s -H 'Content-Type: application/json' -u $ES_USERNAME:$ES_PASSWORD -XGET http://$ES_HOST:$ES_PORT/_snapshot/$SNAPSHOT_NAME/$SNAPSHOT_ID/_status)
+snapshot_state=$(echo $response | jq -r '.snapshots|.[0]|.state')
+
+echo "Snapshot ID: $SNAPSHOT_ID"
```
Comment (aynleslie) [inline elasticsearch-snapshotter/docker/ship.d/run]: nice additions.

File: elasticsearch-snapshotter/docker/ship.d/run (hunk 1 above)
Comment (rlvoyer) [reply elasticsearch-snapshotter/docker/ship.d/run]: Nope. In fact the ES API tells you explicitly to remove it when you pass it. I suspect that the reason is that buckets are not specific to a region.

Comment (rlvoyer) [reply elasticsearch-snapshotter/docker/ship.d/run]: Nope. It's deprecated, and the snapshot API in 5.6.8 will tell you so explicitly. And it makes sense: S3 buckets are not specific to a region.

**Comment by znep** [comment]
I know I'm too late to the party for this PR, but if you are messing around here again then setting `storage_class` to `standard_id`, if it works right, would drop the cost of storing the backups in half.  Not a giant amount, but something.

**Pull request merged**